### PR TITLE
test: remove file-wide interpreter.cpp ubsan suppression

### PR DESCRIPTION
**Pull request by fanquake**

### CODE REVIEW

**Review by hebasto — APPROVED** [review]
ACK 217c0ce552a5d519b5cc702aba0c82514a1c449e.

However, I'd prefer an alternative [approach](https://github.com/bitcoin/bitcoin/pull/29543).

File: test/sanitizer_suppressions/ubsan [hunk 1]
```diff
@@ -55,7 +55,7 @@ unsigned-integer-overflow:MurmurHash3
 unsigned-integer-overflow:CBlockPolicyEstimator::processBlockTx
 unsigned-integer-overflow:TxConfirmStats::EstimateMedianVal
 unsigned-integer-overflow:prevector.h
-unsigned-integer-overflow:script/interpreter.cpp
+unsigned-integer-overflow:EvalScript
```
Comment (Sjors) [inline test/sanitizer_suppressions/ubsan:58]: I assume `stacktop` / `altstacktop` don't work here?

Comment (fanquake) [reply test/sanitizer_suppressions/ubsan:58]: Not in this context, because these are runtime suppressions.

**Comment by fanquake** [comment]
> Is it possible to mark suppressions inside the source code itself?

There are [various ways to suppress issues](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#issue-suppression), however, in this PR, I'm not interested in changing the src, just reducing the scope of the currently applied runtime suppression.

**Review by dergoegge — APPROVED** [review]
ACK 217c0ce552a5d519b5cc702aba0c82514a1c449e

**Comment by fanquake** [comment]
I'm going to merge this to reduce the scope of the suppression, discussion around refactoring / inlining suppressions could can in #29543.

**Pull request merged**